jfrog-cli/artifactory/commands/buildinfo/addartifacts.go

```diff
@@ -0,0 +1,73 @@
+package buildinfo
+
+import (
+	"errors"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/artifactory/utils"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/artifactory/spec"
+	"github.com/jfrog/jfrog-cli-go/jfrog-cli/utils/config"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/utils/errorutils"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/utils/log"
+	"github.com/jfrog/jfrog-cli-go/jfrog-client/artifactory/buildinfo"
+	clientutils "github.com/jfrog/jfrog-cli-go/jfrog-client/artifactory/services/utils"
+)
+
+func AddArtifacts(addArtifactsSpec *spec.SpecFiles, configuration *BuildAddArtifactsConfiguration) (successCount int, failCount int, err error) {
+	log.Info("Running Build Add Artifacts command...")
+	if !configuration.DryRun {
+		buildArtifacts, err := searchArtifacts(configuration.ArtDetails, addArtifactsSpec); if err != nil {
+			return 0, 0, err
+		}
+
+		err = utils.SaveBuildGeneralDetails(configuration.BuildName, configuration.BuildNumber); if err != nil {
+			return 0, len(buildArtifacts), err
+		}
+
+		populateFunc := func(partial *buildinfo.Partial) {
+			partial.Artifacts = buildArtifacts
+		}
+		err = utils.SavePartialBuildInfo(configuration.BuildName, configuration.BuildNumber, populateFunc); if err != nil {
+			return 0, len(buildArtifacts), err
+		}
+
+		return len(buildArtifacts), 0, err
+	} else {
+		return 0, 0, nil
+	}
+}
+
+func searchArtifacts(artifactoryDetails *config.ArtifactoryDetails, addArtifactsSpec *spec.SpecFiles) ([]buildinfo.InternalArtifact, error) {
+	var buildArtifacts []buildinfo.InternalArtifact
+
+	servicesManager, err := utils.CreateServiceManager(artifactoryDetails, false); if err != nil {
+		return nil, err
+	}
+
+	for i := 0; i < len(addArtifactsSpec.Files); i++ {
+		params, err := addArtifactsSpec.Get(i).ToArtifatorySearchParams(); if err != nil {
+			return nil, err
+		}
+		resultItems, err := servicesManager.Search(&clientutils.SearchParamsImpl{ArtifactoryCommonParams: params}); if err != nil {
+			return nil, err
+		}
+		for _, resultItem := range resultItems {
+			artifactPath := resultItem.GetItemRelativePath()
+			artifact := buildinfo.InternalArtifact{Path: artifactPath, Checksum: &buildinfo.Checksum{}}
+			artifact.Sha1 = resultItem.Actual_Sha1
+			artifact.Md5 = resultItem.Actual_Md5
+			buildArtifacts = append(buildArtifacts, artifact)
+		}
+	}
+
+	if len(buildArtifacts) == 0 {
```

Comment:
We should not automatically throw an error here. An error should be thrown if the *fail-no-op* option is set to true. You can use the code in *cli.go* to take care of this for you.
The fact the option does not exist for the *build-add-dependencies* is a bug. We'll soon add it.